pkg/agent/cniserver/interface_configuration_windows.go

```diff
@@ -204,14 +205,27 @@ func (ic *ifConfigurator) removeContainerLink(containerID, epName string) error
 // removeHNSEndpoint removes the HNSEndpoint from HNS and local cache.
 func (ic *ifConfigurator) removeHNSEndpoint(endpoint *hcsshim.HNSEndpoint, containerID string) error {
 	epName := endpoint.Name
+	deleteCh := make(chan error)
 	// Remove HNSEndpoint.
-	_, err := endpoint.Delete()
-	if err != nil {
-		if !strings.Contains(err.Error(), notFoundHNSEndpoint) {
-			klog.Errorf("Failed to delete container interface %s: %v", containerID, err)
-			return err
+	go func() {
+		_, err := endpoint.Delete()
+		deleteCh <- err
+	}()
+
+	// Deleting HNS Endpoint is blocking in some corner cases. It might be a bug in Windows HNS service. To avoid
+	// cniserver is hung, add timeout control in HNSEndpoint deletion.
```

Comment:
s/To avoid cniserver is hung/To avoid hanging in cniserver

Reply:
Updated.